Source/common/SNTConfigurator.m

```diff
@@ -1222,18 +1222,21 @@ - (NSRegularExpression *)expressionForPattern:(NSString *)pattern {
   return [NSRegularExpression regularExpressionWithPattern:pattern options:0 error:NULL];
 }
 
-- (NSMutableDictionary *)readForcedConfig {
-  NSMutableDictionary *forcedConfig = [NSMutableDictionary dictionary];
-  for (NSString *key in self.forcedConfigKeyTypes) {
-    id obj = [self forcedConfigValueForKey:key];
-    forcedConfig[key] = [obj isKindOfClass:self.forcedConfigKeyTypes[key]] ? obj : nil;
-    // Create the regex objects now
-    if (self.forcedConfigKeyTypes[key] == [NSRegularExpression class]) {
-      NSString *pattern = [obj isKindOfClass:[NSString class]] ? obj : nil;
-      forcedConfig[key] = [self expressionForPattern:pattern];
-    }
-  }
+- (void)applyOverrides:(NSMutableDictionary *)forcedConfig {
+  // Overrides should only be applied under debug builds.
 #ifdef DEBUG
+  if ([[[NSProcessInfo processInfo] environment] objectForKey:@"BAZEL_TEST"] &&
```

Comment:
Not sure we need to explicitly check for `BAZEL_TEST`, rather than just having this gated by `ENABLE_CONFIG_OVERRIDES`.

Just using `ENABLE_CONFIG_OVERRIDES` would mean that we'd need to do the following:

It would just mean that we'd need to do either:

* `bazel run --define ENABLE_CONFIG_OVERRIDES=1 //:reload`
* `bazel test --define ENABLE_CONFIG_OVVERRIDES=1 //test:target`

Both of which could be added to the .bazelrc file e.g.

```
# Maintains the old behavior
run:debug --define=ENABLE_CONFIG_OVERRIDES=1
# Gives us a new config to toggle this on / off
run:with_overrides --define ENABLE_CONFIG_OVVERRIDES=1
test:with_overrides --define ENABLE_CONFIG_OVVERRIDES=1
```

So we could do `bazel test --config=with_overrides //test:target`

Reply:
I'm not sure I like this approach as it requires re-compilation to toggle whichever mode you want. It also changes how local debug builds currently work by default which is to apply the overrides, without have another compilation or adding extra build flags.

The current impl has the benefit of not changing the existing workflow for local debug builds (enabling overrides) as well as doing the right thing by default (disabling overrides) when running tests. All without needing recompilation.

Reply:
Yeah as we talked more I agree here. This should be sufficent.